tests/PHPUnit/TestingEnvironment.php

```diff
@@ -256,59 +257,54 @@ public static function addHooks()
             $stylesheets[] = 'tests/resources/screenshot-override/override.css';
         });
-        Piwik::addAction('AssetManager.getJavaScriptFiles', function(&$jsFiles) {
+
+        $globalObservers[] = array('AssetManager.getJavaScriptFiles', function (&$jsFiles) {
             $jsFiles[] = 'tests/resources/screenshot-override/override.js';
         });
-        self::addSendMailHook();
-        Piwik::addAction('Updater.checkForUpdates', function () {
+
+        $globalObservers[] = array('Updater.checkForUpdates', function () {
             try {
                 @\Piwik\Filesystem::deleteAllCacheOnUpdate();
             } catch (Exception $ex) {
                 // pass
             }
         });
-        Piwik::addAction('Platform.initialized', function () use ($testingEnvironment) {
-            static $archivingTablesDeleted = false;
 
-            if ($testingEnvironment->deleteArchiveTables
-                && !$archivingTablesDeleted
-            ) {
-                $archivingTablesDeleted = true;
-                DbHelper::deleteArchiveTables();
-            }
-        });
-        Piwik::addAction('Environment.bootstrapped', function () use ($testingEnvironment) {
-            $testingEnvironment->logVariables();
-            $testingEnvironment->executeSetupTestEnvHook();
-        });
-    }
-
-    public static function addSendMailHook()
-    {
-        Piwik::addAction('Test.Mail.send', function($mail) {
+        $globalObservers[] = array('Test.Mail.send', function (\Zend_Mail $mail) {
             $outputFile = PIWIK_INCLUDE_PATH . '/tmp/' . Common::getRequestVar('module', '') . '.' . Common::getRequestVar('action', '') . '.mail.json';
-
             $outputContent = str_replace("=\n", "", $mail->getBodyText($textOnly = true));
             $outputContent = str_replace("=0A", "\n", $outputContent);
             $outputContent = str_replace("=3D", "=", $outputContent);
-
             $outputContents = array(
-                'from'     => $mail->getFrom(),
-                'to'       => $mail->getRecipients(),
-                'subject'  => $mail->getSubject(),
+                'from' => $mail->getFrom(),
+                'to' => $mail->getRecipients(),
+                'subject' => $mail->getSubject(),
                 'contents' => $outputContent
             );
-
             file_put_contents($outputFile, json_encode($outputContents));
         });
+
+        $globalObservers[] = array('Platform.initialized', function () use ($testingEnvironment) {
+            static $archivingTablesDeleted = false;
+            if ($testingEnvironment->deleteArchiveTables
+                && !$archivingTablesDeleted
+            ) {
+                $archivingTablesDeleted = true;
+                DbHelper::deleteArchiveTables();
+            }
+        });
+
+        $diConfig['observers.global'] = \DI\add($globalObservers);
+
+        StaticContainer::addDefinitions($diConfig);
```

Comment:
`$diConfig` is set to `provideContainerConfig()`, so this would override the entire array (StaticContainer::addDefinitions does not use array_merge()).